Puppet/Puppet/AppDelegate.m

```diff
@@ -226,23 +222,32 @@ - (BOOL)distribute:(MSDistribute *)distribute releaseAvailableWithDetails:(MSRel
 #pragma mark - MSPushDelegate
 
 - (void)push:(MSPush *)push didReceivePushNotification:(MSPushNotification *)pushNotification {
+  NSString *title = pushNotification.title;
   NSString *message = pushNotification.message;
+  NSMutableString *customData = nil;
   for (NSString *key in pushNotification.customData) {
-    message = [NSString stringWithFormat:@"%@\n%@: %@", message, key, [pushNotification.customData objectForKey:key]];
+    ([customData length] == 0) ? customData = [NSMutableString new] : [customData appendString:@", "];
+    [customData appendFormat:@"%@: %@", key, [pushNotification.customData objectForKey:key]];
+  }
+  if (UIApplication.sharedApplication.applicationState == UIApplicationStateBackground) {
+    NSLog(@"%@ Notification received in background, title: \"%@\", message: \"%@\", custom data: \"%@\"", kPUPLogTag,
+          title, message, customData);
+  } else {
+    message = [NSString stringWithFormat:@"%@%@%@", (message ? message : @""), (message && customData ? @"\n" : @""),
```

Comment:
You don't need to check the `message` since it is a mandatory field.

Reply:
A mandatory field yes but not for long ;). Can send silent notifications without title and messages and the backend is getting prepared to that.

Reply:
OK. I've heard about not having title and message in payload. 😄